code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/dao/impl/AuthDaoImpl.java

```diff
@@ -150,7 +187,37 @@ public Account createAccount(String name, String kind, String externalId, String
         if (StringUtils.isNotEmpty(externalType)) {
             properties.put(ACCOUNT.EXTERNAL_ID_TYPE, externalType);
         }
-        return resourceDao.createAndSchedule(Account.class, objectManager.convertToPropertiesFor(Account.class, properties));
+        return resourceDao.createAndSchedule(Account.class, objectManager.convertToPropertiesFor(Account.class,
+                properties));
+    }
+
+    @Override
+    public Credential createUserPassword(Account account, String username, String password) {
+        Map<Object, Object> properties = new HashMap<>();
+        if (StringUtils.isBlank(username) || StringUtils.isBlank(password)) {
+            throw new ClientVisibleException(ResponseCodes.BAD_REQUEST, "UserCredentialCreation", "Cannot have blank username or password.", null);
+        }
+        properties.put(CREDENTIAL.PUBLIC_VALUE, username);
+        properties.put(CREDENTIAL.SECRET_VALUE, password);
+        properties.put(CREDENTIAL.KIND, LocalAuthConstants.PASSWORD);
+        properties.put(CREDENTIAL.ACCOUNT_ID, account.getId());
+        return resourceDao.createAndSchedule(Credential.class, objectManager.convertToPropertiesFor(Credential.class, properties));
+    }
+
+    @Override
+    public Identity getIdentity(Long id) {
+        Account account = getAccountById(id);
+        if (account == null || account.getKind().equalsIgnoreCase(ProjectConstants.TYPE)) {
+            return null;
+        }
+        Credential credential = create()
+                .selectFrom(CREDENTIAL)
+                .where(CREDENTIAL.KIND.equalIgnoreCase("password")
+                        .and(CREDENTIAL.ACCOUNT_ID.eq(id))
+                .and(CREDENTIAL.STATE.equalIgnoreCase(CommonStatesConstants.ACTIVE))).fetchAny();
```

Comment:
Want to make sure I understand this correctly. Here by fetchAny we don't care what identity is returned? So in case of multiple account credentials, it can be any of them? What use case requires this behavior?